File: java/com/github/jasmaa/yahtzee/Yahtzee.java
package com.github.jasmaa.yahtzee;

import java.util.concurrent.ForkJoinPool;

public class Yahtzee {

  private final static int NUM_GAMES = 1000000;
  private final static int NUM_DICE = 5;
  private final static int NUM_TRIES = 3;

  public static void main(String[] args) {
    ForkJoinPool pool = ForkJoinPool.commonPool();
    SimulationTask task = new SimulationTask(NUM_DICE, NUM_TRIES, NUM_GAMES, 10000);
    int numWins = pool.invoke(task);
    System.out.printf("Percentage: %f", (float) numWins / NUM_GAMES);
  }
}

File: kotlin/yahtzee/src/main/kotlin/com/github/jasmaa/yahtzee/Yahtzee.kt
package com.github.jasmaa.yahtzee

private const val NUM_GAMES = 1000000
private const val NUM_DICE = 5
private const val NUM_TRIES = 3

/**
 * Simulates chance of getting yahtzee.
 * This project reuses logic from Java yahtzee.
 */
fun main() {
    var numWins = 0
    for (i in 1..NUM_GAMES) {
        val task = SimulationTask(NUM_DICE, NUM_TRIES, 1, 1)
        numWins += task.compute()
    }
    println("Percentage: ${numWins.toFloat() / NUM_GAMES}")
}